Sources/GRPC/AsyncAwaitSupport/AsyncBidirectionalStreamingCall.swift

```diff
@@ -0,0 +1,166 @@
+  public func cancel() async throws {
+    try await self.call.cancel().get()
+  }
+
+  // MARK: - Response Parts
+
+  /// The initial metadata returned from the server.
+  public var initialMetadata: HPACKHeaders {
+    // swiftformat:disable:next redundantGet
+    get async throws {
+      try await self.responseParts.initialMetadata.get()
+    }
+  }
+
+  /// The trailing metadata returned from the server.
+  public var trailingMetadata: HPACKHeaders {
+    // swiftformat:disable:next redundantGet
+    get async throws {
+      try await self.responseParts.trailingMetadata.get()
+    }
+  }
+
+  /// The final status of the the RPC.
+  public var status: GRPCStatus {
+    // swiftformat:disable:next redundantGet
+    get async {
+      try! await self.responseParts.status.get()
+    }
+  }
+
+  private init(call: Call<RequestPayload, ResponsePayload>) {
+    self.call = call
+    // Initialise `responseParts` with an empty response handler because we
+    // provide the responses as an AsyncSequence in `responseStream`.
+    self.responseParts = StreamingResponseParts(on: call.eventLoop) { _ in }
+
+    // Call and StreamingResponseParts are reference types so we grab a
+    // referecence to them here to avoid capturing mutable self in the  closure
+    // passed to the AsyncThrowingStream initializer.
+    //
+    // The alternative would be to declare the responseStream as:
+    // ```
+    // public private(set) var responseStream: AsyncThrowingStream<ResponsePayload>!
+    // ```
+    //
+    // UPDATE: Additionally we expect to replace this soon with an AsyncSequence
+    // implementation that supports yielding values from outside the closure.
+    let call = self.call
+    let responseParts = self.responseParts
+    let responseStream = AsyncThrowingStream(ResponsePayload.self) { continuation in
+      call.invokeStreamingRequests { error in
+        responseParts.handleError(error)
+        continuation.finish(throwing: error)
+      } onResponsePart: { responsePart in
+        responseParts.handle(responsePart)
+        switch responsePart {
+        case let .message(response): continuation.yield(response)
+        case .metadata: break
+        case .end: continuation.finish()
+        }
+      }
+    }
+    self.responseStream = .init(responseStream)
+  }
+
+  /// We expose this as the only non-private initializer so that the caller
+  /// knows that invocation is part of initialisation.
+  internal static func makeAndInvoke(call: Call<RequestPayload, ResponsePayload>) -> Self {
+    Self(call: call)
+  }
+
+  // MARK: - Requests
+
+  /// Sends a message to the service.
+  ///
+  /// - Important: Callers must terminate the stream of messages by calling `sendEnd()`.
+  ///
+  /// - Parameters:
+  ///   - message: The message to send.
+  ///   - compression: Whether compression should be used for this message. Ignored if compression
+  ///     was not enabled for the RPC.
+  public func sendMessage(
```

Comment:
We should come up with a consistent spelling for sending; on the server I think we talk about "writing" responses rather than sending. 

The same goes for "end" -- for async sequences it seems like "finish" is normal term. 

What do you think about `send(_ request: Request, compression: Compression = .deferToCallDefault)` and `finish()`?